data-source/src/main/scala/tech/ytsaurus/spyt/format/YtOutputWriter.scala

```diff
@@ -151,9 +152,24 @@ class YtOutputWriter(richPath: YPathEnriched,
   protected def initializeWriter(): TableWriter[InternalRow] = {
     val appendPath = richPath.withAttr("append", "true").toYPath
     log.debugLazy(s"Initialize new write: $appendPath, transaction: $transactionGuid")
+    val writeSchemaConverter = WriteSchemaConverter(options)
     val request = WriteTable.builder[InternalRow]()
       .setPath(appendPath)
-      .setSerializationContext(new WriteSerializationContext(new InternalRowSerializer(schema, WriteSchemaConverter(options))))
+      .setSerializationContext(
+        if (options.ytConf(ArrowWriteEnabled)) {
+          if (!writeSchemaConverter.typeV3Format) {
+            throw new RuntimeException("arrow writer is only supported with typeV3")
+          }
+          new ArrowWriteSerializationContext[InternalRow](
+            schema.fields.zipWithIndex.map { case (field, i) =>
+              util.Map.entry(field.name, writeSchemaConverter.ytLogicalTypeV3(field).ytGettersFromStruct(
+                field.dataType, i
+              ))
+            }.toSeq.asJava
+          )
+        } else
```

Comment:
I'd add curly braces for else clause so it would be more explicit where `new WriteSerializationContext` belongs